packages/dev/core/src/Meshes/Compression/dracoDecoder.ts

```diff
@@ -0,0 +1,282 @@
+import { type DecoderModule, DracoDecoderModule } from "draco3dgltf";
+import { DracoCodec, type IDracoCodecConfiguration } from "./dracoCodec";
+import { Tools } from "../../Misc/tools";
+import { Geometry } from "../geometry";
+import { VertexBuffer } from "../buffer";
+import { Logger } from "../../Misc/logger";
+import type { BoundingInfo } from "../../Culling/boundingInfo";
+import type { Scene } from "../../scene";
+import type { Nullable } from "../../types";
+import { decodeMesh, type AttributeData, type Message, workerFunction } from "./dracoCompressionWorker";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare let DracoDecoderModule: DracoDecoderModule;
+
+interface MeshData {
+    indices?: Uint16Array | Uint32Array;
+    attributes: Array<AttributeData>;
+    totalVertices: number;
+}
+
+/**
+ * Draco compression (https://google.github.io/draco/)
+ *
+ * This class wraps the Draco decoder module.
+ *
+ * By default, the configuration points to a copy of the Draco decoder files for glTF from the Babylon.js preview cdn https://preview.babylonjs.com/draco_wasm_wrapper_gltf.js.
+ *
+ * To update the configuration, use the following code:
+ * ```javascript
+ *     DracoDecoder.Config = {
+ *          wasmUrl: "<url to the WebAssembly library>",
+ *          wasmBinaryUrl: "<url to the WebAssembly binary>",
+ *          fallbackUrl: "<url to the fallback JavaScript library>",
+ *     };
+ * ```
+ *
+ * Draco has two versions, one for WebAssembly and one for JavaScript. The decoder configuration can be set to only support WebAssembly or only support the JavaScript version.
+ * Decoding will automatically fallback to the JavaScript version if WebAssembly version is not configured or if WebAssembly is not supported by the browser.
+ * Use `DracoDecoder.Available` to determine if the decoder configuration is available for the current context.
+ *
+ * To decode Draco compressed data, get the default DracoDecoder object and call decodeMeshToGeometryAsync:
+ * ```javascript
+ *     var geometry = await DracoDecoder.Default.decodeMeshToGeometryAsync(data);
+ * ```
+ */
+export class DracoDecoder extends DracoCodec<DecoderModule> {
+    protected static override _Default: Nullable<DracoDecoder> = null;
+    /**
+     * Default instance for the DracoDecoder.
+     */
+    public static get Default(): DracoDecoder {
+        if (!DracoDecoder._Default) {
+            DracoDecoder._Default = new DracoDecoder();
+        }

```

Comment:
nit: nowadays we can write:
```suggestion
        DracoDecoder._Default ||= new DracoDecoder();
```

Reply:
Or rather `??=` in this case since you are really trying to check for null/undefined and not falsy.

Reply:
We could, but I usually try to avoid using `??=` and generally all null coalescing / optional chaining because the downlevel JS is horrendously verbose.